WalletWasabi.Fluent/ViewModels/AddWallet/LegalDocumentsViewModel.cs

```diff
@@ -12,13 +18,46 @@ namespace WalletWasabi.Fluent.ViewModels.AddWallet
 		NavigationTarget = NavigationTarget.DialogScreen)]
 	public partial class LegalDocumentsViewModel : RoutableViewModel
 	{
-		public LegalDocumentsViewModel(string content)
+		[AutoNotify] private string? _content;
+
+		public LegalDocumentsViewModel(LegalChecker legalChecker)
 		{
 			Title = "Terms and Conditions";
-			Content = content;
 			NextCommand = BackCommand;
+			LegalChecker = legalChecker;
+
+			this.WhenAnyValue(x => x.Content)
+				.ObserveOn(RxApp.MainThreadScheduler)
+				.Subscribe(content => IsBusy = content is null);
+		}
+
+		private LegalChecker LegalChecker { get; }
+
+		protected override void OnNavigatedTo(bool inStack, CompositeDisposable disposable)
+		{
+			base.OnNavigatedTo(inStack, disposable);
+
+			Observable.Merge(
+				Observable.FromEventPattern<LegalDocuments>(LegalChecker, nameof(LegalChecker.AgreedChanged)),
+				Observable.FromEventPattern<LegalDocuments>(LegalChecker, nameof(LegalChecker.ProvisionalChanged)))
+				.Select(x => x.EventArgs)
+				.ObserveOn(RxApp.MainThreadScheduler)
+				.Subscribe(legalDocuments => UpdateContent(legalDocuments))
+				.DisposeWith(disposable);
+
+			if (LegalChecker.TryGetNewLegalDocs(out LegalDocuments? provisional))
+			{
+				UpdateContent(provisional);
+			}
+			else if (LegalChecker.CurrentLegalDocument is { } current)
+			{
+				UpdateContent(current);
+			}
 		}
 
-		public string Content { get; }
+		private void UpdateContent(LegalDocuments legalDocuments)
```

Comment:
I would like to separate the update procedure and pass the highest object in the manner of abstraction and update do what it has to do. 

https://github.com/zkSNACKs/WalletWasabi/pull/4995#discussion_r561208807

Reply:
@lontivero What do you think about this?

Reply:
I think it should receive the content. In fact, if you rename `UpdateContent` to `SetContent` (because that's what the method really does) it will be clear that this is the same that `Content.setter` so, the method can be removed and use the setter instead, after all it is a method too.